--- a/cobbler.spec	2018-03-15 06:50:11.595419513 +0000
+++ b/cobbler.spec	2018-03-15 06:58:16.801988290 +0000
@@ -11,11 +11,15 @@
 
 Name:           cobbler
 Version:        2.8.2
-Release:        1%{?dist}
+Release:        2%{?dist}
 Summary:        Boot server configurator
 URL:            http://cobbler.github.io/
 License:        GPLv2+
 Source0:        https://github.com/cobbler/cobbler/archive/v%{version}.tar.gz#/%{name}-%{version}.tar.gz
+Source1:        cobbler-nginx.conf
+Source2:        cobbler-uwsgi-svc.ini
+Source3:        cobbler-uwsgi-web.ini
+Patch0:         cobbler-nginx-%{version}.patch
 
 BuildRequires: git
 BuildRequires: python2-devel
@@ -23,9 +27,12 @@
 BuildRequires: python-cheetah
 BuildRequires: python-setuptools
 
-Requires: httpd
+Requires: nginx
+Requires: uwsgi-plugin-python
+Requires: httpd-tools
+Requires: dhcp
+Requires: pykickstart
 Requires: tftp-server
-Requires: mod_wsgi
 Requires: createrepo
 Requires: python-cheetah
 Requires: python-netaddr
@@ -76,8 +83,6 @@
 %else
 Requires:       Django >= 1.4
 %endif
-Requires:       mod_wsgi
-Requires:       mod_ssl
 Requires(post): openssl
 
 %description -n cobbler-web
@@ -109,6 +115,7 @@
 
 # cobbler
 rm $RPM_BUILD_ROOT%{_sysconfdir}/cobbler/cobbler.conf
+rm $RPM_BUILD_ROOT%{_sysconfdir}/nginx/conf.d/cobbler.conf
 
 mkdir -p $RPM_BUILD_ROOT%{_sysconfdir}/logrotate.d
 mv $RPM_BUILD_ROOT%{_sysconfdir}/cobbler/cobblerd_rotate $RPM_BUILD_ROOT%{_sysconfdir}/logrotate.d/cobblerd
@@ -131,6 +138,11 @@
 
 # cobbler-web
 rm $RPM_BUILD_ROOT%{_sysconfdir}/cobbler/cobbler_web.conf
+rm $RPM_BUILD_ROOT%{_sysconfdir}/nginx/conf.d/cobbler_web.conf
+install -p -m 0644 %{SOURCE1} $RPM_BUILD_ROOT%{_sysconfdir}/nginx/conf.d/cobbler.conf
+mkdir -p $RPM_BUILD_ROOT%{_sysconfdir}/uwsgi.d
+install -p -m 0644 %{SOURCE2} $RPM_BUILD_ROOT%{_sysconfdir}/uwsgi.d/cobbler-svc.ini
+install -p -m 0644 %{SOURCE3} $RPM_BUILD_ROOT%{_sysconfdir}/uwsgi.d/cobbler-web.ini
 
 # koan
 mkdir -p $RPM_BUILD_ROOT/var/spool/koan
@@ -160,7 +172,7 @@
 if (( $1 == 1 )); then
     /sbin/chkconfig --add cobblerd > /dev/null 2>&1
     /etc/init.d/cobblerd start > /dev/null 2>&1
-    /etc/init.d/httpd restart > /dev/null 2>&1
+    /etc/init.d/nginx restart > /dev/null 2>&1
 fi
 %preun
 # before last package is removed
@@ -171,7 +183,7 @@
 %postun
 # after last package is removed
 if (( $1 == 0 )); then
-    /etc/init.d/httpd condrestart > /dev/null 2>&1
+    /etc/init.d/nginx condrestart > /dev/null 2>&1
 fi
 %endif
 
@@ -199,7 +211,6 @@
 %doc AUTHORS README.md docs/README.openvz docs/README.mysql
 %config(noreplace) %{_sysconfdir}/cobbler
 %config(noreplace) %{_sysconfdir}/logrotate.d/cobblerd
-%config(noreplace) /etc/httpd/conf.d/cobbler.conf
 %{_bindir}/cobbler
 %{_bindir}/cobbler-ext-nodes
 %{_bindir}/cobblerd
@@ -221,10 +232,13 @@
 %files -n cobbler-web
 %license COPYING
 %doc AUTHORS README.md
-%config(noreplace) /etc/httpd/conf.d/cobbler_web.conf
-%attr(-,apache,apache) /usr/share/cobbler/web
-%dir %attr(700,apache,root) /var/lib/cobbler/webui_sessions
-%attr(-,apache,apache) /var/www/cobbler_webui_content/
+%attr(-,nginx,nginx) /usr/share/cobbler/web
+%dir %attr(700,nginx,root) /var/lib/cobbler/webui_sessions
+%attr(-,nginx,nginx) /var/www/cobbler_webui_content/
+# configuration
+%config(noreplace) /etc/nginx/conf.d/cobbler.conf
+%config(noreplace) %attr(644,nginx,nginx) /etc/uwsgi.d/cobbler-svc.ini
+%config(noreplace) %attr(644,nginx,nginx) /etc/uwsgi.d/cobbler-web.ini
 
 %files -n koan
 %license COPYING
